### ISSUE TITLE
new DFA object

### ISSUE THREAD

**Issue by chesterlbtan** [kvrigor/dfa2cfg#7]
**Unable to AddState on a newly initialized DFA.**

'_states is NULL at _states.Contains
```
Public Overridable Sub AddStates(ByVal s As String)
            If _states.Contains(s) Then
                Throw New ArgumentException($"The set of states already contains '{s}'.")
            Else
                _states.Add(s)
            End If
End Sub
```

'My sample code in C#
```
myDFA = new Codecs.Languages.Regular.DFA();
            myDFA.InputSymbols = new string[2] { "1", "2" };
            myDFA.AddStates(State.StateCollection.Keys.ToArray());
            foreach (KeyValuePair<string,State> item in State.StateCollection)
            {
                StatesViewer.Rows.Add(item.Key, item.Value.m_TargetOne.Name, item.Value.m_TargetZero.Name);
                Codecs.Languages.Regular.TransFunc trans1 = new Codecs.Languages.Regular.TransFunc();
                trans1.PrevState = item.Key;
                trans1.Input = "1";
                trans1.NextState = item.Value.m_TargetOne.Name;
                myDFA.AddTransitions(trans1);
                Codecs.Languages.Regular.TransFunc trans0 = new Codecs.Languages.Regular.TransFunc();
                trans0.PrevState = item.Key;
                trans0.Input = "0";
                trans0.NextState = item.Value.m_TargetZero.Name;
                myDFA.AddTransitions(trans0);
                if (item.Value.IsAccept)
                    myDFA.AddFinalStates(item.Key);
            }
            Codecs.Languages.ContextFree.CFG sa = Codecs.Utils.DFAExtensions.ToCFG(myDFA, "CFG1");
            richTextBox1.Text = myDFA.ToString();
            richTextBox2.Text = sa.ToString();
```

**Comment by kvrigor** [comment]
fix at 5131415baf114cdf5caf8a4e51449775660c9daa, please verify

if it fixes the issue I'll proceed with PR to master branch

**Comment by kvrigor** [comment]
BTW, use [codecs-bugfixes](/kvrigor/dfa2cfg/tree/codecs-bugfixes) branch to test this
